File: app/src/main/java/org/oppia/app/databinding/GuidelineBinding.java
package org.oppia.app.databinding;

import androidx.constraintlayout.widget.Guideline;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.databinding.BindingAdapter;
import org.jetbrains.annotations.NotNull;

public class GuidelineBinding {
  /** Binding adapter for setting the `guidePercent` for a [Guideline]. */
  @BindingAdapter("app:layout_constraintGuide_percent")
  public static void setGuidelinePercentage(@NotNull Guideline guideline, Float percentage) {
    ConstraintLayout.LayoutParams params = (ConstraintLayout.LayoutParams) guideline.getLayoutParams();
    params.guidePercent = percentage;
    guideline.setLayoutParams(params);
  }
}

File: app/src/main/java/org/oppia/app/databinding/GuidelineBinding.kt
package org.oppia.app.databinding

import androidx.constraintlayout.widget.ConstraintLayout
import androidx.constraintlayout.widget.Guideline
import androidx.databinding.BindingAdapter

/** Binding adapter for setting the `guidePercent` for a [Guideline]. */
@BindingAdapter("app:layout_constraintGuide_percent")
fun setGuidelinePercentage(guideline: Guideline, percentage: Float) {
  val params = guideline.layoutParams as ConstraintLayout.LayoutParams
  params.guidePercent = percentage
  guideline.layoutParams = params
}
